server/controllers/Comments.js

```diff
@@ -107,6 +107,69 @@ class Comments {
       serverError(res);
     }
   }
+
+  /**
+   * @name delete
+   * @async
+   * @static
+   * @memberof Comments
+   * @param {Object} req express request object
+   * @param {Object} res express response object
+   * @returns {JSON} JSON object with message on deleted message
+   */
+  static async delete(req, res) {
+    try {
+      const { slug, id } = req.params;
+      const { id: userId } = req.user;
+      const article = await Article.findBySlug(slug);
+      if (!article) {
+        return serverResponse(res, 404, { error: 'article not found' });
+      }
+      const articleId = article.id;
+      const deletedComment = await Comment.destroy({
+        where: { id, articleId, userId }
+      });
+      if (!deletedComment) {
+        return serverResponse(res, 404, { error: 'comment not found' });
+      }
+      return serverResponse(res, 200, { message: 'comment deleted' });
+    } catch (error) {
+      serverError(res);
+    }
+  }
+
+  /**
+   * @name update
+   * @async
+   * @static
+   * @memberof Comments
+   * @param {Object} req express request object
+   * @param {Object} res express response object
+   * @returns {JSON} JSON object with details of update message
+   */
+  static async update(req, res) {
+    try {
+      const { slug, id } = req.params;
+      const { id: userId } = req.user;
+      let { comment } = req.body;
+      const article = await Article.findBySlug(slug);
+      if (!article) {
+        return serverResponse(res, 404, { error: 'article not found' });
+      }
+      const articleId = article.id;
+      const [rowsUpdate, [updatedComment]] = await Comment.update(
```

Comment:
can you explain what happening at lines 160 to 163?

Reply:
Basically, I am updating a particular comment in the database where the comment Id, user Id and article id is available. 'Returning' is set to true to return an array containing two elements. The first is the number of updated rows while the other is the updated comment details.